README.md

```diff
@@ -1,62 +1,109 @@
+Prerequisites: node, npm and TypeScript (2.4+) are installed.
 
-Add a reference to your target version, e.g.:
+1. Install photoshop.d.ts using `npm` in your project folder:
 
-```typescript
-///<reference path="./node_modules/photoshop.d.ts/dist/cc/ps.d.ts"/>
+```shell
+npm install --save-dev photoshop.d.ts
 ```
 
-The other references will automatically resolve.
+2. Add a `tsconfig.json` to your project with the following recommended
+settings and modify it to your needs: 
+
+```js
+{
+    "compilerOptions": {
+        // Output config
+        "outFile": "./app.jsx",
+        "sourceMap": false,
+        "target": "es3",
+
+        // Module resolution
+        "module": "none",
+
+        // JS handling
+        "allowJs": true,
+        "checkJs": false,
+        "downlevelIteration": true,
+
+        // Type checking
+        "strict": true,
+        "noLib": false,
+        "lib": ["es5"], // exclude DOM and ScriptHost
+        
+        // Include photoshop.d.ts type definitions
+        "types": ["./photoshop.d.ts/dist/cc"]
+        // Alternatively for Photoshop CS6:
+        // "types": ["./photoshop.d.ts/dist/cs6"]
+    },
+     "include": [
+        "./app.ts"
+    ]
+}
+```
 
-## Usage
+Note: If you want to use modules in ExtendScript, you need a module loader like <a href="https://github.com/requirejs/requirejs" target="_blank">requirejs</a> and set the module option to `"amd"`. The setup of requirejs for ExtendScript requires some configuration.
 
-```typescript
-///<reference path="./ps.d.ts"/>
+## ECMAScript version and JavaScript Polyfills
 
-var document = app.activeDocument; // Access global `app`
-var layers   = document.artLayers; // OK
-var bogus    = document.bogus;     // Type Error
-```
+Adobe ExtendScript is based on ECMAScript version 3. Hence the `target` option in the `tsconfig.json` should be set to `es3`. TypeScript will transpile your code into valid ES3 syntax, though it won't transpile any *methods* like ES5 `somearray.forEach` or `.filter` into ES3 for you. 
 
-## Building
+You can use ECMAScript 5 features by adding polyfills to your project. The <a href="https://github.com/inexorabletash/polyfill" target="_blank">es5.js polyfill</a> for example works fine with ExtendScript. To prepend a polyfill to your output file, follow these instructions:
 
-The typescript declarations are generated from the official javascript reference
-PDFs. The text is copied and pasted into local files. The **parser** will parse
-the raw text file and extract meaningful information from the respective
-chapters of the documentation. Once the **parser** has delivered a in-memory
-representation of the types and constants, the **emitter** will render types and
-constants as typescript files.
+1. Download es5.js and place it into your project folder.
+2. Assure `checkJS` ist set to `false` in your `tsconfig.json` when using a 
+JS polyfill library.
+2. Link the polyfill in the `files` section of your `tsconfig.json`: 
```

Comment:
Either increment this to "3." or make all of them "1." to get automatic numbering